sbom-utility-scripts/scripts/add-image-reference-script/add_image_reference.py

```diff
@@ -0,0 +1,404 @@
+        required=True,
+    )
+    parser.add_argument(
+        "--input-file",
+        "-i",
+        type=Path,
+        help="SBOM file in JSON format.",
+        required=True,
+    )
+    parser.add_argument(
+        "--output-file",
+        "-o",
+        type=str,
+        help="Path to save the output SBOM in JSON format.",
+    )
+    return parser
+
+
+def update_component_in_cyclonedx_sbom(sbom: dict, image: Image) -> dict:
+    """
+    Update the CycloneDX SBOM with the image reference.
+
+    The reference to the image is added to the SBOM in the form of a component and
+    purl is added to the metadata.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        image (Image): An instance of the Image class that represents the image.
+
+    Returns:
+        dict: Updated SBOM with the image reference added.
+    """
+    # First, update the metadata with the image purl
+    sbom["metadata"]["component"]["purl"] = image.purl()
+
+    # Then, add the image component to the components list
+    image_component = {
+        "type": "container",
+        "name": image.name,
+        "purl": image.purl(),
+        "version": image.tag,
+        "hashes": [{"alg": image.digest_algo_cyclonedx, "content": image.digest_hex_val}],
+    }
+    sbom["components"].insert(0, image_component)
+    sbom["metadata"]["component"] = image_component
+    return sbom
+
+
+def find_package_by_spdx_id(sbom: dict, spdx_id: str) -> Optional[dict]:
+    """
+    Find the package in the SBOM by SPDX ID.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        spdx_id (str): SPDX ID of the package to find.
+
+    Returns:
+        dict: The package with the given SPDX ID.
+    """
+    for package in sbom["packages"]:
+        if package["SPDXID"] == spdx_id:
+            return package
+    return None
+
+
+def delete_package_by_spdx_id(sbom: dict, spdx_id: str) -> dict:
+    """
+    Delete the package in the SBOM by SPDX ID.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        spdx_id (str): SPDX ID of the package to delete.
+
+    Returns:
+        dict: Updated SBOM with the package deleted.
+    """
+    for package in sbom["packages"]:
+        if package["SPDXID"] == spdx_id:
+            sbom["packages"].remove(package)
+            break
+    return sbom
+
+
+def redirect_virtual_root_to_new_root(sbom: dict, virtual_root: str, new_root: str) -> dict:
+    """
+    Redirect the relationship describing the document to a new root node.
+
+    Args:
+        sbom (dict): SBOM in JSON format.
+        virtual_root (str): A virtual root node identifier that needs to be replaced.
+        new_root (str): A new root node identifier that will replace the virtual root node.
+
+    Returns:
+        dict: Updated SBOM with the virtual root node replaced.
+    """
+    for relationship in sbom["relationships"]:
+        if relationship["spdxElementId"] == virtual_root:
+            relationship["spdxElementId"] = new_root
+
```

Comment:
Should we do this for the `relatedSpdxElement` as well? The virtual root can also be on the "right side" of relationships, e.g. `some_package BUILD-TOOL-OF virtual_root`

Reply:
I added a logic that covers an other direction.